File: algorithms/python/Challenge_Permutation_Sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        retStr = ""
        perm = 1
        setList = [i for i in range(1, n+1)]
        
        for i in range(1, n):
            perm *= i
        
        for i in range(n-1, 0, -1):
            retStr += str(setList.pop((k-1)//perm))
            k = k - perm*((k-1)//perm)
            perm //= i
        
        retStr += str(setList.pop(0))
        
        return retStr

File: algorithms/java/Challenge_Permutation_Sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        List<Integer> permList = new ArrayList<>(n);
        int perm = 1;
        String retStr = "";
        
        for(int i = 1 ; i < n ; ++i) {
            permList.add(i);
            perm *= i;
        }
        permList.add(n);
        
        for(int i = n-1 ; i > 0 ; --i) {
            retStr += Integer.toString(permList.remove((k-1)/perm));
            k -= perm*((k-1)/perm);
            perm /= i;
        }
        retStr += Integer.toString(permList.remove(0));
        return retStr;
    }
}
